### ISSUE TITLE
ArrayIndexOutOfBoundsException in tryCity method from maxmind

### ISSUE THREAD

**Issue by edelgadoh** [maxmind/MaxMind-DB-Reader-java#68]
Dear maxmind developers,

We are using the last lib 

       <dependency>
            <groupId>com.maxmind.geoip2</groupId>
            <artifactId>geoip2</artifactId>
            <version>2.13.0</version>
        </dependency>

From time to time we observed some errors from maxmind lib, you can see the stacktrace below:

`"stack_trace":"java.lang.ArrayIndexOutOfBoundsException: Index 34 out of bounds for length 16\n\tat com.maxmind.db.Decoder$Type.get(Decoder.java:53)\n\tat com.maxmind.db.Decoder.decode(Decoder.java:129)\n\tat com.maxmind.db.Decoder.decode(Decoder.java:88)\n\tat com.maxmind.db.Reader.resolveDataPointer(Reader.java:256)\n\tat com.maxmind.db.Reader.getRecord(Reader.java:176)\n\tat com.maxmind.geoip2.DatabaseReader.get(DatabaseReader.java:271)\n\tat com.maxmind.geoip2.DatabaseReader.tryCity(DatabaseReader.java:334)`

As we analysed the code, the error starts in Decoder.class, specifically in this part:

   static enum Type {
        EXTENDED,
        POINTER,
        UTF8_STRING,
        DOUBLE,
        BYTES,
        UINT16,
        UINT32,
        MAP,
        INT32,
        UINT64,
        UINT128,
        ARRAY,
        CONTAINER,
        END_MARKER,
        BOOLEAN,
        FLOAT;

        static final Decoder.Type[] values = values();

        private Type() {
        }

        static Decoder.Type get(int i) {
            return values[i];
        }

sometimes the **i** parameter is *36*, but there are only 16 fixed values in the ENUM.

Thanks

**Comment by edelgadoh** [comment]
The exact mechanism used in (3) is:

We used JMeter to make a GET request to one fixed endpoint that uses MaxMind (using SpringBoot and Java Amazon Corretto 11), we configured JMeter to create 100 threads, making an interval of 5 seconds to wait and repeat it 3 times. The all process takes for about 15 seconds. 

During this 15 seconds of execution, we replaced between the old and new MaxMind databases (using the terminal we made a cp -r /tmp/oldMaxmind/ /project/maxmind-files/ and then a cp -r /tmp/newMaxmind/ /project/maxmind-files/ at least 6 times)

My notebook is an Ubuntu 19: 
Linux NI-45848-1 5.3.0-050300-generic #201909152230 SMP Sun Sep 15 22:32:54 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

Thank you!
